interceptors/interceptors.go

```diff
@@ -137,6 +139,16 @@ func ServerErrorInterceptor() grpc.UnaryServerInterceptor {
 	}
 }
 
+//GRPCRecoveryInterceptor recover if server panic, return error with panic stack
+func GRPCRecoveryInterceptor() grpc.UnaryServerInterceptor {
+	opts := []grpc_recovery.Option{
+		grpc_recovery.WithRecoveryHandler(func(p interface{}) (err error) {
```

Comment:
hi, can you help try if we can move https://github.com/carousell/Orion/blob/master/orion/handlers/http/http.go#L39 to here?

Reply:
^ @ankurs can you take a look?

Reply:
got it, will check the http part.

Reply:
Hi @hothero, I've checked the content of http handler part

getHTTPHandler
-> httpHandler: handle new relic, add log (k, v) to context, panic recover, notifier
	-> serveHTTP
		-> 1. handle usual http handler, set by 
			`func (h *httpHandler) AddHTTPHandler(serviceName string, method string, path string, handler handlers.HTTPHandler)`
		   2. http handler carried with info.svc.svc
		   3. grpc converted to http request, only this type will apply interceptors we defined in DefaultInterceptors()

If we remove the implementation
https://github.com/carousell/Orion/blob/master/orion/handlers/http/http.go#L39 
for case 1, 2, there will be no protection, monitoring on them, only case 3 we will cover because of the interceptors.

I think the better way is for GRPCRecoveryInterceptor(), it should let httpHandler to protect itself and add the check to ignore http request.
`
if modifiers.IsHTTPRequest(ctx) {
			return handler(ctx, req)
		}
`

Reply:
Could you help test this branch? https://github.com/carousell/Orion/pull/105 If it covers those cases you memtioned

Reply:
ok, will check